Bioinf_Utility/modules/fastq_utils.py

```diff
@@ -0,0 +1,149 @@
+    """
+    Read a single sequence from a FASTQ file.
+    """
+    at_header_line = input_file.readline().strip()
+    sequence_line = input_file.readline().strip()
+    input_file.readline().strip()
+    quality_line = input_file.readline().strip()
+
+    seq_name = at_header_line.strip().split(" ")[0]
+
+    return seq_name, sequence_line, quality_line
+
+
+def write_fastq_dict_to_file(sequences, output_fastq):
+    """
+    Write a dictionary of sequences to a FASTQ file.
+    """
+    with open(output_fastq, "w") as output_file:
+        for seq_name, (sequence, quality_seq) in sequences.items():
+            write_fastq_seq_to_file(
+                seq_name, sequence, quality_seq, output_file
+            )
+
+
+def write_fastq_seq_to_file(seq_name, sequence, quality_seq, output_file):
+    """
+    Write a single sequence to a FASTQ file.
+    """
+    output_file.writelines(
+        [
+            seq_name + "\n",
+            sequence + "\n",
+            "+" + seq_name[1:] + "\n",
+            quality_seq + "\n",
+        ]
+    )
+
+
+def convert_multiline_fasta_to_oneline(
+    input_fasta_path, output_fasta_path=None
+):
+    """
+    Convert a multi-line FASTA file to a single-line format.
+    """
+    if output_fasta_path is None:
+        filename, extension = os.path.splitext(input_fasta_path)
+        output_fasta_path = filename + "_oneline_format" + extension
+
+    with (
+        open(input_fasta_path, "r") as input_file,
+        open(output_fasta_path, "w") as output_file,
+    ):
+        sequence_lines = []
+        for line in input_file:
+            line = line.strip()
+            if line.startswith(">"):
+                if sequence_lines:
+                    output_file.write("".join(sequence_lines) + "\n")
+                    sequence_lines = []
+                output_file.write(line + "\n")
+            else:
+                sequence_lines.append(line)
+
+        if sequence_lines:
+            output_file.write("".join(sequence_lines) + "\n")
+
+
+def parse_blast_output(input_file_path, output_file_path):
+    """
+    Parse a BLAST output file to extract best matches.
+    """
+    best_matches = set()
+
+    with open(input_file_path, "r") as input_file:
+        for line in input_file:
+            line = line.strip()
+
+            if line.startswith("Sequences producing significant alignments:"):
+                input_file.readline()
+                header_line = input_file.readline().strip()
+                name_column_start_index = header_line.find("Name")
+                best_match = (
+                    input_file.readline()
+                    .strip()[:name_column_start_index]
+                    .strip()
+                )
+                best_matches.add(best_match)
+
+    sorted_best_matches = sorted(best_matches)
+
+    with open(output_file_path, "w") as output_file:
+        for description in sorted_best_matches:
+            output_file.write(description + "\n")
```

Comment:
Нет проверок, что output_file_path существует. Можно либо добавить выскакивание ошибки, либо задать путь внутри кода. Типа так: 

```suggestion
    if output_file_path == '':
        directory = input_file_path.rsplit('/', 1)[0]
        output_file_name = 'parse_blast.txt'
        output_file = f"{directory}/{output_file_name}"
```